src/metaprob/state.clj

```diff
@@ -137,12 +136,12 @@
 (defn map-to-state [m]
   (let [n (count m)]
     (cond (and (= n 2)
-               (not (= (get m :value :no-value) :no-value))
-               (seq? (get m rest-marker :no-value)))
+               (contains? m :value)
+               (seq? (get m rest-marker)))
           (cons (get m :value)
                 (get m rest-marker))
 
-          (= n 0) '()                   ;Kludge to ensure seq-ness
+          (zero? n) '()                   ;Kludge to ensure seq-ness
```

Comment:
Do we really want to be checking for real and complex zero here?  We know it's an integer. (I guess a compiler would already know that; but someone reading the code would have to infer it.  Just recording my thought process here)